test/rest-api/applications.js

```diff
@@ -1,58 +1,56 @@
 const should = require('should');
 const idGen = require('uuid62');
 const adminHelper = require('../common/admin-helper')();
+const username = idGen.v4();
 
 describe('REST: Applications', () => {
```

Comment:
Can you add the same tests for users? 
And I've seen the following behavior:
count=100 actually returns 14 users. Each user has 7 fields so it is 98 <100;  user 15 is on the next page
Interesting that app tests seem to behave differently. you request count 3 and check for 3 apps

Is it the difference between real redis vs mock or apps are implemented differently than users?

Reply:
Sure, I'm going to replicate the test.

Regarding the count — it seems to be a bit complicated.

 I've been digging into the docs this morning and it seems to be working a little bit differently compared to what you said yesterday (or maybe I didn't get what you said correctly).

From Redis Docs:

> It is possible to empirically adjust the behavior of SCAN using the COUNT option

> Basically with COUNT the user specified the amount of work that should be done at every call in order to retrieve elements from the collection. This is just a hint for the implementation.

So the `count` parameter is not really about "how big you want the resultset to be" but rather "how many keys I'm going to process per each iteration"

I've also checked the `ioredis-mock` implementation for the `scan` command and what they do is
1. Fetch **all** the keys
2. Select a window of `count` keys
3. Filter the result using the match statement

So — if in the particular range there are no matching keys, you'll get an empty result, which is what the docs is stating as well: 

> The commands are also allowed to return zero elements, and the client should not consider the iteration complete as long as the returned cursor is not zero.

Also — but I might be wrong here — the `scan` command is iterating exclusively on all the keys in the database, it never fetches their values. So I'm not sure how the number of field that a key has is involved in the discussion.

Does that clarify?

Reply:
aha, that is interesting. so some pages might be empty. looks good for now